9.x ui tests for the "nic attributes" feature

rkhozinov opened the pull request.

Commit:
9.x ui tests for the "nic attributes" feature

naumvd95 commented on servers/partner-ci/9.x/ui.yaml:
```diff
@@ -0,0 +1,233 @@
+#---
```
why yaml-symbol commented?

rkhozinov replied:
fixed

naumvd95 commented on servers/partner-ci/9.x/ui.yaml:
```diff
@@ -0,0 +1,233 @@
+    testrail_plan: "${{TREP_TESTRAIL_MILESTONE}} snapshot #${{SNAPSHOTS_ID}}"
+    testrail_plan_desc: "Test plan description"
+    testrail_run: "${{TREP_TESTRAIL_SUITE}}"
+    testrail_run_desc: "Test run description"
```
seems we need to redefine this vars directly

naumvd95 commented on servers/partner-ci/9.x/ui.yaml:
```diff
@@ -0,0 +1,233 @@
+          submodule:
+            disable: true
+            tracking: true
+            recursive: true
```
im not sure thats multiple scm will work well

rkhozinov replied:
it works properly, I've already tested it

naumvd95 replied:
ok

naumvd95 commented on servers/partner-ci/builders/ui.test.sh:
```diff
@@ -0,0 +1,42 @@
+[[ "${DEBUG}" == "true" ]] && set -x || set +x
+
+fuel_release=$(echo $ISO_FILE | cut -d- -f2 | tr -d '.iso')
+export FUEL_RELEASE=$fuel_release
```
maybe one-line defining?

rkhozinov replied:
missed check.
fixed

naumvd95 commented on servers/partner-ci/common.yaml:
```diff
@@ -129,6 +129,93 @@
             WORKSTATION_USERNAME=vmware;
 
 - parameter:
+    name: common-fuel-parameters
```
that looks like rewriting, seems to me -bad case, need to discuss

rkhozinov replied:
We need to avoid of passing unused parameters.
We should facilitate "clear fuel" jobs.

naumvd95 replied:
ok, need to restructure it in future

naumvd95 commented on servers/partner-ci/common.yaml:
```diff
@@ -330,6 +417,28 @@
           fi
 
```
strange rows

rkhozinov replied:
see code from 398 to 417

naumvd95 replied:
ok

naumvd95 commented on servers/partner-ci/common.yaml:
```diff
@@ -330,6 +417,28 @@
+      - shell: |
+          #!/bin/bash
+          cherrypick(){
+              git_url=$1
```
we need a uniform case for getting refs
but your function is unique
need to discuss

rkhozinov replied:
it's needed to avoid code duplication

rkhozinov replied:
It seems to be refactored in the next future

naumvd95 replied:
can u create issue to remember that?

naumvd95 commented on servers/partner-ci/common.yaml:
```diff
@@ -395,12 +506,13 @@
-                [[ -z "${SNAPSHOTS_ID}" ]] && { echo "SNAPSHOTS_ID is empty"; exit 1; }
-                wget --no-check-certificate -O ${SNAPSHOT_ARTIFACTS_FILE} \
-                ${SNAPSHOTS_URL/SNAPSHOTS_ID/$SNAPSHOTS_ID}
+                #!/bin/bash -xeu
```
seems to me , previous case was more properly

rkhozinov replied:
{ echo "SNAPSHOTS_ID is empty"; exit 1; } is equal to {SNAPSHOTS_ID:?}

naumvd95 replied:
ok

Commit:
9.x.ui code-review fixes

Commit:
tox: ui suite is added

Commit:
common: fuel_qa_branch for ui tests is changed

Commit:
common: style fix

Commit:
9.x/ui: fuel env shouldn't be erased

Commit:
9.x/ui: branch and refs is added

Commit:
9.x/ui: TEST_PREFIX parameters is added

Commit:
9.x/ui: scm changes

 - http://git.* requires default choosing-strategy
 - scm worspace should be erased before clone

Commit:
ui.test: docker-compose related changes

naumvd95 committed:
reporter for ui-job

Commit:
9.x/ui: unsubscribe ekhokmyakova

Commit:
9.x/ui: fix ui-publisher

Commit:
9.x/ui: reporter.properties generation is added

Commit:
common_prepare: force_reuse should stop pip upgrade

rkhozinov committed:
Merge branch 'master' into ui-tests

naumvd95 committed:
fix getting-snapshot case for specific ui tests

rkhozinov committed:
Merge pull request #111 from rkhozinov/pipnoupdate

common_prepare: force_reuse should stop pip upgrade
